sequence_processing_pipeline/NuQCJob.py

```diff
@@ -396,6 +398,42 @@ def _process_sample_sheet(self):
                 'projects': lst,
                 'sample_ids': sample_ids}
 
+    def _generate_mmi_filter_cmds(self, mmi_db_paths, working_dir):
+        initial_input = join(working_dir, "seqs.r1.fastq")
+        final_output = join(working_dir, "seqs.r1.ALIGN.fastq")
+
+        cmds = []
+
+        tmp_file1 = join(working_dir, "foo")
+        tmp_file2 = join(working_dir, "bar")
+
+        for count, mmi_db_path in enumerate(mmi_db_paths):
+            if count == 0:
+                # prime initial state with unfiltered file and create first of
+                # two tmp files.
+                input = initial_input
+                output = tmp_file1
+            elif count % 2 == 0:
+                # alternate between two tmp file names so that the input and
+                # output are never the same file.
+                input = tmp_file2
+                output = tmp_file1
+            else:
+                input = tmp_file1
+                output = tmp_file2
+
+            cmds.append(f"minimap2 -2 -ax sr -t 16 {mmi_db_path} {input} -a |"
```

Comment:
Here the 16 in both commands means that is requesting 16 threads each so if the script is requesting 4 {cores_per_task} (per default), then you are over subscribing. The solution is to divide by 2 and assign each; something like this:
```python
alignment_filter_cores = int(cores_per_task/2)
cmds.append(f"minimap2 -2 -ax sr -t {alignment_filter_cores} {mmi_db_path} {input} -a |" ...
```